plugins/inputs/neptune_apex/neptune_apex.go

```diff
@@ -0,0 +1,265 @@
+	HardwareVersion string   `xml:"hardware,attr"`
+	Hostname        string   `xml:"hostname"`
+	Serial          string   `xml:"serial"`
+	Timezone        float64  `xml:"timezone"`
+	Date            string   `xml:"date"`
+	PowerFailed     string   `xml:"power>failed"`
+	PowerRestored   string   `xml:"power>restored"`
+	Probe           []probe  `xml:"probes>probe"`
+	Outlet          []outlet `xml:"outlets>outlet"`
+}
+
+type probe struct {
+	Name  string  `xml:"name"`
+	Value string  `xml:"value"`
+	Type  *string `xml:"type"`
+}
+
+type outlet struct {
+	Name     string  `xml:"name"`
+	OutputID string  `xml:"outputID"`
+	State    string  `xml:"state"`
+	DeviceID string  `xml:"deviceID"`
+	Xstatus  *string `xml:"xstatus"`
+}
+
+// NeptuneApex implements telegraf.Input.
+type NeptuneApex struct {
+	Servers         []string
+	ResponseTimeout internal.Duration
+	httpClient      *http.Client
+}
+
+// Description implements telegraf.Input.Description
+func (*NeptuneApex) Description() string {
+	return "Neptune Apex data collector"
+}
+
+// SampleConfig implements telegraf.Input.SampleConfig
+func (*NeptuneApex) SampleConfig() string {
+	return `
+  ## The Neptune Apex plugin reads the publicly available status.xml data from a local Apex.
+  ## Measurements will be logged under "apex".
+
+  ## The hostname/IP of the local Apex(es). If you specify more than one server, they will
+  ## be differentiated by the "hostname" tag.
+  servers = [
+    "apex.local",
+  ]
+
+  ## The response_timeout specifies how long to wait for a reply from the Apex.
+  #response_timeout = "5s"
+`
+}
+
+// Gather implements telegraf.Input.Gather
+func (n *NeptuneApex) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+	for _, server := range n.Servers {
+		wg.Add(1)
+		go func(server string) {
+			defer wg.Done()
+			acc.AddError(n.gatherServer(acc, server))
+		}(server)
+	}
+	wg.Wait()
+	return nil
+}
+
+func (n *NeptuneApex) gatherServer(acc telegraf.Accumulator, server string) error {
+	resp, err := n.sendRequest(server)
+	if err != nil {
+		return err
+	}
+	return n.parseXML(acc, resp)
+}
+
+// parseXML is strict on the input and does not do best-effort parsing. This is because of the life-support nature
+// of the Neptune Apex.
+func (n *NeptuneApex) parseXML(acc telegraf.Accumulator, data []byte) error {
+	r := xmlReply{}
+	err := xml.Unmarshal(data, &r)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal XML: %v\nXML DATA: %q", err, data)
+	}
+
+	var reportTime time.Time
+	if reportTime, err = parseTime(r.Date, r.Timezone); err != nil {
+		return err
+	}
+
+	mainFields := map[string]interface{}{
+		"software":       r.SoftwareVersion,
+		"hardware":       r.HardwareVersion,
+		"serial":         r.Serial,
+		"timezone":       r.Timezone,
+		"power_failed":   r.PowerFailed,
+		"power_restored": r.PowerRestored,
+	}
+	acc.AddFields(Measurement, mainFields, map[string]string{"hostname": r.Hostname, "type": "controller"}, reportTime)
```

Comment:
Can you line wrap around around 78 chars through the pr?

Reply:
Done.
I'm assuming 78 + \r + \n == the golden 80 char/line?
I would recommend adding this requirement to the CONTRIBUTING.md file as it's usually not a requirement for Go projects (And the tools aren't built for it).

Reply:
78 is just what I have my text editor set to, I'm not sure why, maybe I read [rfc2822](https://tools.ietf.org/html/rfc2822#section-2.1.1) at some point, anyway the exact number is not super important for me but its helpful to be able to see the full line without move horizontally and I often have several files up side by side.

I added some info about this to the various plugin pages: 1d6ff4fe4c3b535da56565a3271aece463123794